### ISSUE TITLE
Симулятор. 

### ISSUE THREAD

**Issue by zelezomen** [SBProX/SBProX-Dev#482]
Радует скорость выхода обновлений, надеюсь таким темпом и до симулятора доберётесь. Понимаю, что функционал не первой необходимости, но хотелось бы видеть его в рабочем виде, а не как приписку, что он есть.
Если нужно, я разобью всё по отдельным задачам, но чтобы сим стал юзабельным, делать нужно всё и выкатывать новый плеер со всеми доработками. Что-то уже выкладывалось, но был ответ "отложено", возможно время пришло, очень неудобно скакать от терминала к терминалу - реалтайм в одном, плеер в другом. 

1) При запуске плеера меняется масштаб графика по вертикали. Нужно, чтобы он оставался рабочим, тем, который был до запуска симулятора.

2) Нужен выбор времени начала проигрывания истории. Если торговать только американскую сессию, то каждый раз приходится на ускорении проматывать до открытия основной сессии, неудобно.

3) Нужны настраиваемые кнопки скорости воспроизведения, чтобы на горячую, без паузы, можно было менять ускорение. Хотя бы 3-4 штуки, чтобы была возможность настроить, например, x2, x5, x100 и x1000 или другие варианты, кому как удобно. Или сделать функционал по перемотке на N минут, как в тайгере.

4) Плашки ордеров сделать меньше и/или убрать их с поля графика. Можно сделать как в нинзе, когда можно указанием смещения плашки от правого края убрать её к левому. Можно сделать как в волфиксе, когда ордера на шкале цены, а на линии только подписи.

5) Сделать возможным продлевать линию цены ордера либо на весь чарт либо на какую-то величину в пикселях. 

6) Дать возможность убрать подписи к отметкам трейдов на чарте. Если одна сделка за сессию, то ещё ок, но если есть пара перезаходов, доборы или выходы частями, то график не видно за этими подписями. 

7) Нужна фиксация статистики. Сейчас результат есть только с момента запуска плеера и до его закрытия, при новом запуске всё обнуляется. Нужен выбор или возможность создания аккаунта для симулятора, и чтобы результаты сохранялись на дистанции, чтобы можно было отработать длительный период времени и видеть результат в виде статистики, графика эквити и конкретных сделок.

8) Нужна какая-то кнопка выхода из режима симулятора, чтобы по одному клику всё возвращалось к исходному состоянию. 

9) Нужна реальная динамика изменения цены. Сейчас всё происходит раз в секунду или около того, что не даёт никакого представления об агрессии участников и вообще о  состоянии сил, действующих на цену. Что ETH, что основная сессия - всё одно, будто нет никого. Цена как бы меняется, но это совсем не то, что происходит в реалтайм.

**Comment by bushevigor** [comment]
Доработайте побыстрей симулятор

**Comment by rvasilchenko** [comment]
> Радует скорость выхода обновлений, надеюсь таким темпом и до симулятора доберётесь. Понимаю, что функционал не первой необходимости, но хотелось бы видеть его в рабочем виде, а не как приписку, что он есть.
> Если нужно, я разобью всё по отдельным задачам, но чтобы сим стал юзабельным, делать нужно всё и выкатывать новый плеер со всеми доработками. Что-то уже выкладывалось, но был ответ "отложено", возможно время пришло, очень неудобно скакать от терминала к терминалу - реалтайм в одном, плеер в другом.
> 
> 1. При запуске плеера меняется масштаб графика по вертикали. Нужно, чтобы он оставался рабочим, тем, который был до запуска симулятора.
> 2. Нужен выбор времени начала проигрывания истории. Если торговать только американскую сессию, то каждый раз приходится на ускорении проматывать до открытия основной сессии, неудобно.
> 3. Нужны настраиваемые кнопки скорости воспроизведения, чтобы на горячую, без паузы, можно было менять ускорение. Хотя бы 3-4 штуки, чтобы была возможность настроить, например, x2, x5, x100 и x1000 или другие варианты, кому как удобно. Или сделать функционал по перемотке на N минут, как в тайгере.
> 4. Плашки ордеров сделать меньше и/или убрать их с поля графика. Можно сделать как в нинзе, когда можно указанием смещения плашки от правого края убрать её к левому. Можно сделать как в волфиксе, когда ордера на шкале цены, а на линии только подписи.
> 5. Сделать возможным продлевать линию цены ордера либо на весь чарт либо на какую-то величину в пикселях.
> 6. Дать возможность убрать подписи к отметкам трейдов на чарте. Если одна сделка за сессию, то ещё ок, но если есть пара перезаходов, доборы или выходы частями, то график не видно за этими подписями.
> 7. Нужна фиксация статистики. Сейчас результат есть только с момента запуска плеера и до его закрытия, при новом запуске всё обнуляется. Нужен выбор или возможность создания аккаунта для симулятора, и чтобы результаты сохранялись на дистанции, чтобы можно было отработать длительный период времени и видеть результат в виде статистики, графика эквити и конкретных сделок.
> 8. Нужна какая-то кнопка выхода из режима симулятора, чтобы по одному клику всё возвращалось к исходному состоянию.
> 9. Нужна реальная динамика изменения цены. Сейчас всё происходит раз в секунду или около того, что не даёт никакого представления об агрессии участников и вообще о  состоянии сил, действующих на цену. Что ETH, что основная сессия - всё одно, будто нет никого. Цена как бы меняется, но это совсем не то, что происходит в реалтайм.

Согласен. Плюс хотелось бы симулятор не только на тиках, но и просто на отдельных барах c шагом по кнопке. И, конечно, чтоб график был полнофункциональным и на нем можно было рисовать фигуры для анализа при этом.

**Comment by mosnfet** [comment]
1. Хотелось бы чтоб задавалось время обновления, одно в обновление в секунду, два в секунду и тд. И наоборот 1 обновление в 2 секунды и тд.
2. И перемотка с выпадающем списком на столько то баров назад и в другую сторону - настолько то баров вперед. Допустим 1,5, 10, 15, 25, 50.
